staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/validation.go

```diff
@@ -308,11 +308,11 @@ func validateNestedValueValidation(v *NestedValueValidation, skipAnyOf, skipAllO
 	if len(v.ForbiddenExtensions.XListMapKeys) > 0 {
 		allErrs = append(allErrs, field.Forbidden(fldPath.Child("x-kubernetes-list-map-keys"), "must be empty to be structural"))
 	}
-	if v.ForbiddenExtensions.XListType != nil {
-		allErrs = append(allErrs, field.Forbidden(fldPath.Child("x-kubernetes-list-type"), "must be undefined to be structural"))
+	if len(v.ForbiddenExtensions.XListType) != 0 {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("x-kubernetes-list-type"), "must be atomic to be structural"))
 	}
-	if v.ForbiddenExtensions.XMapType != nil {
-		allErrs = append(allErrs, field.Forbidden(fldPath.Child("x-kubernetes-map-type"), "must be undefined to be structural"))
+	if len(v.ForbiddenExtensions.XMapType) != 0 {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("x-kubernetes-map-type"), "must be granular to be structural"))
```

Comment:
same here, if they set it to "granular", won't it still fail validation with a message saying it must be "granular"?